templates/debugging.md

```diff
@@ -0,0 +1,52 @@
+# Debugging the Provider
```

Comment:
Why have you introduced this template? `debugging.md` is not templated docs and not “part” of the provider as such - it is simply some information to allow anyone developing the provider to debug it.
If all docs need to become templates for some reason then you will need  to add something to the docs explaining this - otherwise someone may contribute another `.md` file, or indeed may make changes to the version in the docs folder because there does not seem to be any reason for the file to be templated.

If you want (or need) the files to be templates then I think you should add a [markdown comment](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#hiding-content-with-comments) at the top of every generated file so that any reader of the docs who may be considering editing them will know to edit the template and not the doc they are reading

Reply:
I added this template to maintain the position of `debuggin.md` in the current document structure. 

The generated files all have the metadata at the top containing `# generated by https://github.com/hashicorp/terraform-plugin-docs`.

perhaps I misunderstood the premise of this comment, please let me know if there's something else?

Reply:
In the first instance I was wondering why this needs to be a template at all and could not just have stayed where it was and not be a template at all.
The comment that says it is auto generated is all well and good…but generated from where?
Think about a new contributor to the repo -they spot a spelling mistake in `debugging.md` - the comment tells them this is autogenerated..but where do they go to find the original and update it?
Or someone wants to add a new documentation file to describe something else…where do they put that file?  If it needs to be a template, is it enough that it exists in the template folder or does something else need to happen?
What if they just put it in the folder they want it to appear in?

It is not particularly intuitive to have to go looking for templates for files that are essentially stand-alone documentation - if it is strictly necessary for them to be templates then you must consider from the point of view of someone who neither knows nor cares about the templating system - what is the least friction you can cause them to achieve what they want?   Guiding them where necessary and automating (as per my suggestion on generating docs at PR time rather than forcing the contributor to do a manual run) where possible helps to reduce friction

Reply:
Got you now @freeranger. I've changed this approach up to have `debugging.md` just remain in the `docs` directory, rather than being cloned as part of the generation. As such, I've also actioned @jdamata's point about the `rm` commands in the `make docs` target. This was why I had debugging in the templates - as make docs essentially removed the docs dir and generates each time to ensure they are up to date (slightly heavy handed, perhaps). I've tested now with the `debugging.md` in `/docs` and the target without `rm` commands and all seems well.

Reply:
Would you prefer I did the same with `index.md` @freeranger/ @jdamata  ? It also doesn't have any auto-generated content, as I just took the existing index file as it was.

Reply:
So `index.md` does need templated to maintain consistency with the existing docs - the generated docs lose some elements.

The difference with the header is that `debugging.md` isn't actually part of the 'site'. If you look on the provider docs it doesn't appear (https://registry.terraform.io/providers/jdamata/sonarqube/latest/docs). It seems it was moved into the `/docs` directory last year, but by default Terraform provider docs don't expect this. I could put it into `/docs/guides/debugging.md` with the metadata header, then it would show on the site. But is this what you want? Or should this debugging info live outside of `/docs` at the root of the repo like a `contributing.md` would?

Reply:
Ok, fair enough about `index.md` 
I don’t think `debugging.md` needs to be part of the provider docs since it only applies to debugging the provider and not to using it.
Whether it should stay where it is and stay like it is or move to another folder I will leave to @jdamata.
I am happy to leave it where it is but don’t feel strongly enough about the location.
Thanks

Reply:
Thakns @freeranger for the feedback. Will hang off to see what @jdamata's thoughts are and work on the other tidy ups in the meantime.

Reply:
Im cool with moving it to the root of the repo.